client/src/main/java/io/pravega/client/connection/impl/TcpClientConnection.java

```diff
@@ -117,31 +121,52 @@ public void run() {
             IoBuffer buffer = new IoBuffer();
             while (!stop.get()) {
                 try {
+                    isReadingFromSocket.set(true);
+                    // This method blocks until it is able to read data from the tcp socket InputStream.
                     WireCommand command = readCommand(in, buffer);
+                    isReadingFromSocket.set(false);
```

Comment:
What happens if the line above throws? Should we reset this value to `false` in a `finally` block?

Reply:
An exception from `readCommand(..)` method will be caught and the Socket inputStream will be closed . Also, a concurrent `stop()` invocation will be a NOP. Hence I do not see a need for the finally block here.